config/config.go

```diff
@@ -60,6 +59,57 @@ type SessionsConfig struct {
 	MaxPerUser uint32
 }
 
+// Counter for the limits  and restrictions for the File Transfer
+//on and off the device(MEN-4325)
+type Counters struct {
+	// Maximum bytes count allowed to transfer per hours
+	// this is per device global limit, which is consulted
+	// every time there is a transfer starting. if above
+	// the limit, we answer with error message indicating
+	// limit reached.
+	MaxBytesTxPerHour uint64
+	MaxBytesRxPerHour uint64
+}
+
+// Limits and restrictions for the File Transfer on and off the device(MEN-4325)
+type FileTransferLimits struct {
+	// the global parent directory that File Transfer will never escape
+	Chroot string
+	// No way to escape Chroot, even if this one is set the Chroot setting will
+	// be checked for the target of any link and restricted accordingly
+	FollowSymLinks bool
+	// Allow overwrite files
+	AllowOverwrite bool
+	// set the owner of new files to OwnerPut
+	OwnerPut string
+	// set the owner of new files to OwnerPut
+	GroupPut string
+	// allow to get only files owned by OwnerGet
+	OwnerGet string
+	// allow to get only files owned by OwnerGet
+	GroupGet string
+	// umask for new files
+	Umask string
+	// Maximum allowed file size
+	MaxFileSize uint64
+	// Global counters
+	Counters Counters
+	// If true it is allowed to upload files with set user id on execute bit set
+	AllowSuid bool
+	// By default we only allow to send/put regular files
+	RegularFilesOnly bool
```

Comment:
I guess this only affects stat requests? Uploads and downloads are already restricted to regular files.

Reply:
really? I tried with socket and fifo, did not dare to try char/block /dev'ices but I think this limit makes sense to have explicitly.

Reply:
I was wrong, it does affect downloads. Uploads are however restricted to regular files because of this line: https://github.com/mendersoftware/mender-connect/blob/1399abbbb0478ecc91e197b5ae2dbcc719236f7d/session/filetransfer.go#L471
And the way the file is created:
https://github.com/mendersoftware/mender-connect/blob/1399abbbb0478ecc91e197b5ae2dbcc719236f7d/session/filetransfer.go#L397